src/PAK-Samples/MapReferenceToPaper/Program.cs

```diff
@@ -0,0 +1,415 @@
+                }
+
+                // Reduce the mapping down to only the most confident ones based on inverse Jaccard distance
+                var confidentMappings = new List<Tuple<double, string, string, string>>();
+                foreach (var b in entityAttributeMapping.OrderByDescending(a => a.Item1).ThenByDescending(a => a.Item4.Length))
+                {
+                    if (b.Item1 > 0.8 && reducedChunk.Contains(b.Item4))
+                    {
+                        confidentMappings.Add(b);
+                        reducedChunk = reducedChunk.Replace(b.Item4, "");
+                    }
+                }
+
+                // For each mapping we're confident with, generate a mapping string with embedded XML tags for each match
+                foreach (var b in confidentMappings.OrderByDescending(a => a.Item1).ThenByDescending(a => a.Item4.Length))
+                {
+                    if (b.Item3 == b.Item4)
+                    {
+                        expandedChunk = Regex.Replace(expandedChunk, $"({Regex.Escape(b.Item4)})(?![^<]*>|[^<>]*</)", $" <attr confidence=\"{b.Item1}\" name=\"academic#{b.Item2}\">{b.Item4}</attr> ");
+                    }
+                    else
+                    {
+                        expandedChunk = Regex.Replace(expandedChunk, $"({Regex.Escape(b.Item4)})(?![^<]*>|[^<>]*</)", $" <attr confidence=\"{b.Item1}\" name=\"academic#{b.Item2}\" canonical=\"{b.Item3}\">{b.Item4}</attr> ");
+                    }
+                }
+
+                remainingUnmatched += (remainingUnmatched == "" ? "" : " ") + reducedChunk;
+                newParse = Regex.Replace(newParse, $"({Regex.Escape(unmatchedChunk)})(?![^<]*>|[^<>]*</)", expandedChunk);
+
+                // Collapse whitespace down to single space
+                remainingUnmatched = Regex.Replace(remainingUnmatched, "\\s+", " ").Trim();
+                newParse = Regex.Replace(newParse, "\\s+", " ").Trim();
+
+                var response = new ReferenceMapping
+                {
+                    PercentOfReferenceMapped = 1.0 - ((double)remainingUnmatched.Length / (double)interpretedQuery.Length),
+                    NormalizedReference = interpretedQuery,
+                    MappedPaper = entity,
+                    MappedReference = newParse,
+                    OriginalReference = referenceString,
+                    ReferenceNotMapped = remainingUnmatched
+                };
+
+                return response;
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Generates interpretation of query using Project Academic Knowledge Interpret method
+        /// </summary>
+        /// <param name="query">Natural language query to interpret</param>
+        /// <returns>JSON string of the Interpret method response. See https://docs.microsoft.com/en-us/academic-services/project-academic-knowledge/reference-interpret-method</returns>
+        static string GetQueryInterpretations(
```

Comment:
>GetQueryInterpretations [](start = 22, length = 23)

Wouldn't these requests get throttled eventually?

Reply:
Yes

---
In reply to: [395925547](https://github.com/microsoft/mag-covid19-research-examples/pull/11#discussion_r395925547) [](ancestors = 395925547)

Reply:
Should we self throttle then?

---
In reply to: [395927328](https://github.com/microsoft/mag-covid19-research-examples/pull/11#discussion_r395927328) [](ancestors = 395927328,395925547)

Reply:
They will eventually get throttled by the 10k/month limit, but stopping them before that would be the same as them hitting the limit unless I'm missing your point..

---
In reply to: [395927503](https://github.com/microsoft/mag-covid19-research-examples/pull/11#discussion_r395927503) [](ancestors = 395927503,395927328,395925547)

Reply:
I guess the question is, can someone run this script w/o hitting the 10K limit?  If so, we should probably point out that out.

---
In reply to: [395932827](https://github.com/microsoft/mag-covid19-research-examples/pull/11#discussion_r395932827) [](ancestors = 395932827,395927503,395927328,395925547)

Reply:
The sample data has ~2k rows, so yes

---
In reply to: [395935469](https://github.com/microsoft/mag-covid19-research-examples/pull/11#discussion_r395935469) [](ancestors = 395935469,395932827,395927503,395927328,395925547)

Reply:
Also, the current version of MAKES does not support the new default grammar yet, so we unfortunately can't point them to it just yet

---
In reply to: [395935469](https://github.com/microsoft/mag-covid19-research-examples/pull/11#discussion_r395935469) [](ancestors = 395935469,395932827,395927503,395927328,395925547)